pkg/versionstream/versionstreamrepo/gitrepo.go

```diff
@@ -131,27 +79,51 @@ func cloneJXVersionsRepo(versionRepository string, versionRef string, settings *
 			log.Logger().Infof(util.QuestionAnswer("A local Jenkins X versions repository already exists, pulling the latest", util.YesNo(pullLatest)))
 		}
 		if pullLatest {
-			w, err := repo.Worktree()
-			if err == nil {
-				err := w.Pull(&git.PullOptions{RemoteName: "origin"})
+			err = gitter.FetchRemoteTags(wrkDir, versionRepository)
+			if err != nil {
+				dir, err := deleteAndReClone(wrkDir, versionRepository, versionRef, gitter)
 				if err != nil {
-					dir, err := deleteAndReClone(wrkDir, versionRepository, versionRef, gitter)
-					if err != nil {
-						return "", "", errors.WithStack(err)
-					}
-					return dir, versionRef, nil
+					return "", "", errors.WithStack(err)
 				}
+				return dir, versionRef, nil
 			}
-		}
-		if versionRef != "" {
-			err = gitter.Checkout(wrkDir, versionRef)
+			err = gitter.FetchBranch(wrkDir, versionRepository, versionRef)
 			if err != nil {
 				dir, err := deleteAndReClone(wrkDir, versionRepository, versionRef, gitter)
 				if err != nil {
 					return "", "", errors.WithStack(err)
 				}
 				return dir, versionRef, nil
 			}
+			// TODO: It might be good to do the same behavior for any branch - just the checkout approach works for
+			// tags and commit-ishs, but not branches. I'm not sure how we can tell if it's a branch vs a tag, though.
```

Comment:
The only way I know how to do this is to use the FilterTags method to check if the tag exists or not